src/app/util/attribute-table.h

```diff
@@ -17,14 +17,60 @@
 
 #pragma once
 
+#include <app/ConcreteAttributePath.h>
 #include <app/util/af-types.h>
 #include <app/util/attribute-metadata.h>
 #include <lib/core/DataModelTypes.h>
 #include <protocols/interaction_model/StatusCode.h>
 
+#include <cstdint>
+
+struct EmberAfWriteDataInput
```

Comment:
Suggest `EmberAfWriteParameters`

Reply:
I considered this, however `Parameters` generally sounds like the whole set of parameters, however our write functions generally have a path which I kept split as I was unclear if we actually do want a full parameters only structure here. So the logic was we have sparate "where to write" (path) and "what to write" (which is this input), although I do agree it is somewhat odd that this "what" also contains information about how to mark change and where to mark the changes callbacks.